webdriver-extension/index.bs

```diff
@@ -0,0 +1,532 @@
+=====================
+
+## Commands ## {#mock-sensor-commands}
+
+### Create mock sensor ### {#create-mock-sensor-command}
+
+<table>
+  <tbody>
+    <tr>
+      <th>HTTP Method</th>
+      <th><a lt="extension command URI template">URI Template</a></th>
+    </tr>
+    <tr>
+      <td>POST</td>
+      <td>/session/{session id}/sensor</td>
+    </tr>
+  </tbody>
+</table>
+
+The <dfn>Create mock sensor</dfn> <a>extension command</a> simulates user
+creation of a <a>mock sensor</a>.
+
+The <a>remote end steps</a> are:
+1. Let |configuration| be the |configuration| argument, <a>converted to an IDL value</a>
+   of type {{MockSensorConfiguration}}. If this throws an exception, return a
+   <a>WebDriver error</a> with <a>WebDriver error code</a> <a>invalid argument</a>.
+2. Let |type| be the |configuration|.{{MockSensorConfiguration/mockSensorType}}, if <a>current browsing context</a>
+   already exists this |type| of [=mock sensor=], return a <a>WebDriver error</a> with <a>WebDriver error code</a>
+   [=mock sensor already created=].
+3. Run these sub-steps [=in parallel=] to create a [=mock sensor=] in <a>current browsing context</a>:
+  1. Set the [=mock sensor type|mock sensor's type=] to |type|.
+  2. Let |connected| be the |configuration|.{{MockSensorConfiguration/sensorShouldConnected}},
+     force the result of invoking <a>connect to sensor</a> with associated <a>Sensor</a> instance as argument to |connected|.
+  3. If |configuration|.{{MockSensorConfiguration/maxSamplingFrequency}} is [=present=], then
+    1. Set the [=mock sensor=]'s associated maximum supported sampling frequency to |configuration|.{{MockSensorConfiguration/maxSamplingFrequency}}.
+  4. If |configuration|.{{MockSensorConfiguration/minSamplingFrequency}} is [=present=], then
+    1. Set the [=mock sensor=]'s minimum supported sampling frequency to |configuration|.{{MockSensorConfiguration/minSamplingFrequency}}.
+4. Return <a>success</a> with data `null`.
+
+<div class="example">
+  To create an "ambient-light" mock sensor of <a>current browsing context</a> of the <a>session</a> with ID 23,
+  the <a>local end</a> would POST to `/session/23/sensor/` with the body:
+  <pre class="lang-json">
+  {
+    "mockSensorType": "ambient-light",
+    "maxSamplingFrequency": 60,
+    "minSamplingFrequency": 5
+  }
+  </pre>
+</div>
+
+### Get mock sensor ### {#get-mock-sensor-command}
+
+<table>
+  <tbody>
+    <tr>
+      <th>HTTP Method</th>
+      <th><a lt="extension command uri template">URI Template</a></th>
+    </tr>
+    <tr>
+      <td>GET</td>
+      <td>/session/{session id}/sensor/{type}</td>
+    </tr>
+  </tbody>
+</table>
+
+The <dfn>Get mock sensor</dfn> <a>extension command</a> simulates user
+getting information of a given type of <a>mock sensor</a>.
+
+The <a>remote end steps</a> are:
+1. Let |type| be a <a>url variable</a>.
+2. If {{MockSensorType}} [=set/contains|does not contain=] |type|, return a <a>WebDriver error</a> with
```

Comment:
I fear that https://infra.spec.whatwg.org/#list-contain isn't the right operation for checking if a string is an enum member. Maybe something like trying to convert the string to an IDL value, or check if WebDriver itself does this somewhere. I'd ask @automatedtester if in doubt.

Reply:
Hmm, I am not quite sure either, but use "convert the string to an IDL value" would be more appropriate.